Samples/WinMLSamplesGallery/WinMLSamplesGalleryNative/External/CSampleQueue.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <windows.h>
+#include <Mfidl.h>
+#include "common.h"
+#include <mutex> 
+
+using namespace MediaFoundationSamples;
+
+class CSampleQueue: public IUnknown
+{
+public:
+    class   ILockedSample;
+
+    static  HRESULT Create(
+                            CSampleQueue**   ppQueue
+                            );
+
+#pragma region IUnknown
+    // IUnknown Implementations
+    ULONG   __stdcall   AddRef();
+    HRESULT __stdcall   QueryInterface(
+                                REFIID  riid,
+                                void**  ppvObject
+                                );
+    ULONG   __stdcall   Release();
+#pragma endregion IUnknown
+
+    // ILockedSampleCallback Implementation
+            
+            HRESULT AddSample(                      // Add a sample to the back of the queue
+                            IMFSample*  pSample
+                            );
+            HRESULT GetNextSample(                  // Remove a sample from the front of the queue
+                            IMFSample** pSample
+                            );
+            HRESULT RemoveAllSamples();
+            HRESULT MarkerNextSample(
+                            const ULONG_PTR pulID
+                            );
+            bool    IsQueueEmpty();
+
+            ULONG GetLength();
+protected:
+                class CNode;
```

Comment:
Nit: Just curious about the indentation here since it doesn't seem to match the indentation underneath the public section